pkg/types/indicator.go

```diff
@@ -408,6 +408,41 @@ func Array(a Series, limit ...int) (result []float64) {
 	return
 }
 
+type RegressionType string
+
+const (
+	RegressionTypeConstant              RegressionType = "c"
+	RegressionTypeConstantTrend         RegressionType = "ct"
+	RegressionTypeConstantTrendSeasonal RegressionType = "ctt"
+	RegressionTypeNone                  RegressionType = "n"
+)
+
+// Ordinary Least Squares fit result, only support 1d array
+func OSL(a SeriesExtend, b SeriesExtend, n int) (float64, float64) {
+	if a.Length() < n {
+		return 0, 0
+	}
+	if b.Length() < n {
+		return 0, 0
+	}
+	numerator := 0.0
```

Comment:
Before starting the calculation, maybe it's better to add a check on the equality of the length of `a` and `b`?

Reply:
Not necessary. We only process last n elements of the time series.